### PR TITLE
Timeseries SQL source

### LINKED ISSUE #467: Implement SqlTimeSeriesSource and SqlTimeSeriesMappingSource
Analogous to `CsvTimeSeriesSource` and `CsvTimeSeriesMappingSource`.

- [x] #473

### PR DESCRIPTION
**Pull request by sebastian-peter**
Resolves #467

### CODE REVIEW

**Comment by sebastian-peter** [comment]
Note: I'm aware that SonarQube does not like `SqlTimeSeriesSource.java#L50`, but I am not aware of a solution that does not throw such a warning.

**Comment by sebastian-peter** [comment]
Note: I'm aware that SonarQube does not like `SqlTimeSeriesSource.java#L50`, but I do not know a solution that does not throw such a warning.

**Comment by sebastian-peter** [comment]
!test

File: src/main/java/edu/ie3/datamodel/io/naming/FileNamingStrategy.java [hunk 1]
```diff
@@ -242,56 +236,16 @@ public FileNameMetaInformation extractTimeSeriesMetaInformation(String fileName)
...
-      String fileName) {
-    Matcher matcher = getLoadProfileTimeSeriesPattern().matcher(fileName);
-    if (!matcher.matches())
-      throw new IllegalArgumentException(
-          "Cannot extract meta information on load profile time series from '" + fileName + "'.");
-
-    return new LoadProfileTimeSeriesMetaInformation(
-        UUID.fromString(matcher.group("uuid")), matcher.group("profile"));
-  }
-
```
Comment (t-ober) [inline src/main/java/edu/ie3/datamodel/io/naming/FileNamingStrategy.java:294]: What is the reason of moving this to `EntityPersistenceNamingStrategy` ?

File: src/main/java/edu/ie3/datamodel/io/naming/FileNamingStrategy.java [hunk 2]
```diff
@@ -311,8 +265,7 @@ public String getIdCoordinateEntityName() {
   public Optional<String> getIdCoordinateFilePath() {
     // do not adapt orElseGet, see https://www.baeldung.com/java-optional-or-else-vs-or-else-get for
     // details
-    return getFilePath(
-        getIdCoordinateEntityName(), fileHierarchy.getBaseDirectory().orElseGet(() -> ""));
+    return getFilePath(getIdCoordinateEntityName(), fileHierarchy.getBaseDirectory().orElse(""));
```
Comment (t-ober) [inline src/main/java/edu/ie3/datamodel/io/naming/FileNamingStrategy.java:268]: As for the comment above this shouldn't be changed. Any reason you did do it nevertheless?

**Review by t-ober — CHANGES_REQUESTED** [review]
Nice stuff, just minor things. 
Not sure if it makes sense if someone else looks over this since I'm not too familiar with all that I/O PSDM stuff.

File: src/main/java/edu/ie3/datamodel/io/naming/FileNamingStrategy.java (hunk 1 above)
Comment (sebastian-peter) [reply src/main/java/edu/ie3/datamodel/io/naming/FileNamingStrategy.java:294]: `FileNamingStrategy` was constructed (as the name hints at) for file-related functionality. 
Timeseries meta information needed regardless of whether the data stems from files or sql, thus I moved it to `EntityPersistenceNamingStrategy`. This way, sql sources do not requrie a `FileNamingStrategy` for initialization.

File: src/main/java/edu/ie3/datamodel/io/naming/FileNamingStrategy.java [hunk 3]
```diff
@@ -97,8 +91,8 @@ public FileNamingStrategy() {
     // do not adapt orElseGet, see https://www.baeldung.com/java-optional-or-else-vs-or-else-get for
     // details
     return getFilePath(
-        entityPersistenceNamingStrategy.getEntityName(timeSeries).orElseGet(() -> ""),
-        getDirectoryPath(timeSeries).orElseGet(() -> ""));
+        entityPersistenceNamingStrategy.getEntityName(timeSeries).orElse(""),
+        getDirectoryPath(timeSeries).orElse(""));
```
[the remark this answers is not in the document]
Comment (sebastian-peter) [reply src/main/java/edu/ie3/datamodel/io/naming/FileNamingStrategy.java:95]: Good catch

[the remark this answers is not in the document]
Comment (sebastian-peter) [reply src/main/java/edu/ie3/datamodel/io/naming/FileNamingStrategy.java:95]: Reversed it and also adapted other code lines accordingly

**Review by t-ober — APPROVED** [review]

**Pull request merged**